Resources/js/translator.js

```diff
@@ -53,6 +53,17 @@ var Translator = (function(document, undefined) {
         var _locale = locale || currentLocale || localeFallback,
             _domain = domain;
 
+        /**
+         * If the _locale looks like nl_NL or nl-NL
+         * Try the language as the locale
+         */
+        if (undefined == _messages[_locale]) {
+            var tryLanguage = _locale.match(/^([a-z]{2})(_|-)[A-Z]{2}$/)
```

Comment:
Symfony actually support any number of sublocales, not only one.

And you should not restrict it to 2 letters IMO (especially not enforcing the case), as it is inconsistent with Symfony.